src/main/scala/is/hail/methods/VEP.scala

```diff
@@ -56,6 +56,30 @@ object VEP {
     (Locus(a(0), a(1).toInt), a(3) +: a(4).split(","))
   }
 
+  def getCSQHeaderDefinition(cmd: Array[String], confEnv: Map[String, String]): Option[String] = {
+    val csqHeaderRegex = "ID=CSQ[^>]+Description=\"([^\"]+)".r
+    val pb = new ProcessBuilder(cmd.toList.asJava)
+    val env = pb.environment()
+    confEnv.foreach { case (key, value) => env.put(key, value) }
+    
+    val (jt, proc) = List((Locus("1", 13372), IndexedSeq("G", "C"))).iterator.pipe(pb,
+      printContext,
+      printElement,
+      _ => ())
+
+    val csqHeader = jt.flatMap(s => csqHeaderRegex.findFirstMatchIn(s).map(m => m.group(1)))
+    val rc = proc.waitFor()
+    if (rc != 0)
+      fatal(s"VEP command failed with non-zero exit status $rc")
+
+    if (csqHeader.hasNext)
```

Comment:
Just to confirm -- this is checking if the csqHeader is empty or not (i.e. match found) and you use hasNext and next because jt is an iterator with the VEP result.

Reply:
csqHeader is an interator of all lines that match the header regex in the vep result. This takes the first one.